### PR TITLE
gateway-api: add Gateway API instructions to egress gateways doc

### PR DESCRIPTION
**Pull request by frankbu**
## Reviewers

- [ ] Ambient
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure
- [ ] Localization/Translation

### CODE REVIEW

File: content/en/docs/tasks/traffic-management/egress/egress-gateway/index.md [hunk 1]
```diff
@@ -135,80 +135,161 @@ First create a `ServiceEntry` to allow direct traffic to an external service.
...
+{{< tab name="Gateway API" category-value="gateway-api" >}}
+
+{{< text bash >}}
+$ kubectl apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: cnn-egress-gateway
+  annotations:
+    networking.istio.io/service-type: ClusterIP
```
Comment (frankbu) [inline content/en/docs/tasks/traffic-management/egress/egress-gateway/index.md:186]: This seems like the only "special" config needed to create an egress gateway (i.e., no public IP as usual for ingress gateways). Is there anything else that should be configured differently for egress vs. ingress gateways?

cc @howardjohn

Comment (frankbu) [reply content/en/docs/tasks/traffic-management/egress/egress-gateway/index.md:186]: Should there be a non Istio-specific way to indicate or deduce a Gateway is for egress vs ingress?

Comment (howardjohn) [reply content/en/docs/tasks/traffic-management/egress/egress-gateway/index.md:186]: This looks right to me. Egress and ingress are, IMO, the same thing. You could have 1 gateway doing ingress and egress. Splitting them is mostly an operational convention (and you generally don't want external access to egress)

**Review by ericvn — APPROVED** [review]

**Pull request merged**